### PR TITLE
Publish macOS builds

### PR DESCRIPTION
**Pull request by tellowkrinkle**
### Description of Changes
Updates Mac build script to make publishable builds and enables build publishing

Note: Don't merge until after #5792, #5807, and #5844 are merged

### Rationale behind Changes
Official Mac builds

### Suggested Testing Steps
Not sure how to test the publishing system without merging to master, if anyone knows how please test

### CODE REVIEW

File: .github/workflows/macos-workflow.yml
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:141]:
```suggestion
            if [ -n "${{ github.event.pull_request.title }}" ]; then
```

File: .github/workflows/macos-workflow.yml
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:136]:
```suggestion
            if [ -n "${{ github.event.pull_request.number }}" ]; then
```

File: .github/workflows/macos-workflow.yml
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:132]:
```suggestion
```

File: .github/workflows/macos-workflow.yml
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:130]: This is the default shell.
```suggestion
        shell: bash
```

**Comment by refractionpcsx2** [comment]
All the above are merged, so you can merge this when you're ready to publish them.

File: .github/workflows/macos-workflow.yml
Comment (xTVaser) [inline .github/workflows/macos-workflow.yml:166]: i believe this is going to make the release asset poorly named -- in the linux/windows steps there is an additional step that names them cleaner than the current PR/Commit artifact https://github.com/PCSX2/pcsx2/blob/master/.github/workflows/windows-workflow.yml#L188-L194

Additionally, should the name of the artifact include the version of macOS that is supported? I see the CI is using big sur.

Also this can be tested on a fork before merged, i can do so.

File: .github/workflows/macos-workflow.yml
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:164]:
```suggestion
          path: *.tar.gz
```

File: .github/workflows/macos-workflow.yml
Comment (Margen67) [inline .github/workflows/macos-workflow.yml:174]:
```suggestion
          cd .github/workflows/scripts/releases/upload-release-artifacts
```

File: .github/workflows/macos-workflow.yml
Comment (tellowkrinkle) [reply .github/workflows/macos-workflow.yml:166]: The whole point of doing this is to make the build support all the way back to our minimum version (10.13 at the moment).  If we wanted to only support Big Sur and up, we could continue using Homebrew binaries.

I name it `macOS.tar.gz` since we only support one architecture and one SIMD ISA.  If that changes in the future, we could rename it, though ideally we'd make a fat binary and shove it all in one download.  Does that seem reasonable?

Comment (xTVaser) [reply .github/workflows/macos-workflow.yml:166]: The name of the asset will be used to make it clear to the user on the frontend what version it supports.  I'm not saying it should be labelled as big sur (although if something is ever merged that breaks on 10.13, aka high sierra?, it's probably going to go unnoticed).

If you are targetting high sierra, then we should label it as such.

Comment (tellowkrinkle) [reply .github/workflows/macos-workflow.yml:166]: The compiler is pretty good about making sure we keep compatibility, ignoring 10.13-specific bugs, so I don't expect compatibility breaks to happen very often.

If we're going to label macOS 10.13, should we making the Windows one "Windows 8.1"?

Comment (xTVaser) [reply .github/workflows/macos-workflow.yml:166]: Yes I've thought about that argument.  From my experience with macOS apps, they usually label what version they support clearly -- I suppose this has something to do with how frequently macOS versions break previous app compatibility, but it seems to be the norm.

I guess its ultimately up to you, I have no experience building for macOS.  But the current naming this PR will commit will break the backing API's expectations as seen above -- I imagined that we would we labelling it with the version given my personal experience.

Comment (tellowkrinkle) [reply .github/workflows/macos-workflow.yml:166]: I looked around at some other applications.  Chrome shows versions for both.  Opera does for neither.  Dolphin does for neither.  Sublime Text does for neither.  VSCode does for neither.  IntelliJ does neither.

What application were you thinking of that only shows a version for the macOS release?

Comment (xTVaser) [reply .github/workflows/macos-workflow.yml:166]: Basically almost everyone I can remember downloading for my own mac 🤷 clearly it's not as standardized as I thought.

**Review by xTVaser — CHANGES_REQUESTED** [review]
The `paths-ignore` here needs to be removed, and the workflow needs to run on tag pushes for the process to work https://github.com/PCSX2/pcsx2/pull/5848/files#diff-e24d0c032fd3ded7d8b2f5704cf5e79e808d7577926e58a1f9334c51c9c48decR9

That was the only change i had to make here - https://github.com/xTVaser/pcsx2-rr/commit/558f2c786836d2fa29ee12361a684df137f33960

Which was able to create a successful release - https://github.com/xTVaser/pcsx2-rr/releases/tag/v1.4.2

**Review by xTVaser — APPROVED** [review]
This looks good and should work after merging.  I'll follow up on the rest of the work to get the mac builds pushed through the rest of the process (on the downloads page).

**Comment by tellowkrinkle** [comment]
Okay.  I just realized I need to make sure we put up a decent error if you try to run on a non-Metal-supporting mac, so I won't merge quite yet

**Pull request merged**
